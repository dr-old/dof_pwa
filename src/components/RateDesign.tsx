import { Box, Divider, IconButton, Typography, useTheme } from "@mui/material";
import { hexToRgba } from "../utils/helpers";
import { ArrowOutwardRounded } from "@mui/icons-material";
import { useState } from "react";

const settings = {
  width: 170,
  height: 170,
  value: 60,
};

interface ArcDesignProps {
  backgroundColor?: string;
  title?: string;
}

interface DetailProps {
  backgroundColor?: string;
  title?: string;
  value?: string;
}

const Detail = ({ backgroundColor, title, value }: DetailProps) => {
  const theme = useTheme();
  const color =
    theme.palette.mode === "light"
      ? theme.palette.text.primary
      : theme.palette.background.default;

  return (
    <Box
      sx={{
        justifyContent: "space-between",
        alignItems: "center",
        display: "flex",
        flexDirection: "row",
      }}>
      <Typography
        variant="body2"
        sx={{
          color,
          display: "flex",
          justifyContent: "center",
          alignItems: "center",
        }}>
        <Box
          sx={{
            width: 12,
            height: 12,
            backgroundColor,
            borderRadius: "30%",
            marginRight: 1,
          }}
        />
        {title}
      </Typography>
      <Typography variant="body2" sx={{ color }}>
        {value}
      </Typography>
    </Box>
  );
};

export default function RateDesign({ backgroundColor, title }: ArcDesignProps) {
  const theme = useTheme();
  const [isSelected, setSelected] = useState<number | null>(null);
  const color =
    theme.palette.mode === "light"
      ? theme.palette.text.primary
      : theme.palette.background.default;

  return (
    <Box
      sx={{
        backgroundColor,
        padding: theme.spacing(2),
        borderRadius: theme.shape.borderRadius,
        textAlign: "center",
        minHeight: 320,
        display: "flex",
        flex: 1,
        flexDirection: "column",
      }}>
      <Box
        sx={{
          display: "flex",
          flexDirection: "row",
          alignItems: "center",
          justifyContent: "space-between",
        }}>
        <Typography variant="subtitle1" fontWeight={600} sx={{ color }}>
          {title}
        </Typography>
        <IconButton
          sx={{
            width: 35, // Adjust the size as needed
            height: 35, // Adjust the size as needed
            display: "flex",
            alignItems: "center",
            justifyContent: "center",
            borderRadius: "50%", // Fully rounded
            backgroundColor: color, // Light background for better visibility
            color: theme.palette.secondary.main,
          }}>
          <ArrowOutwardRounded fontSize="small" color="inherit" />
        </IconButton>
      </Box>
      <Box
        sx={{
          height: 200, // Adjust the height of the chart
          justifyContent: "center",
          alignItems: "center",
          display: "flex",
          columnGap: 2,
        }}>
        {[
          { title: "Mon", value: 2 },
          { title: "Tue", value: 3 },
          { title: "Wed", value: 4 },
        ].map((item, index) => {
          return (
            <Box
              key={index.toString()}
              onClick={() => setSelected(item.value)}
              sx={{
                width: 70,
                height: 70,
                borderWidth: 1,
                display: "flex",
                alignItems: "center",
                justifyContent: "center",
                borderColor:
                  isSelected === item.value
                    ? theme.palette.warning.main
                    : hexToRgba(color, 0.3),
                borderRadius: 3,
                backgroundColor:
                  isSelected === item.value
                    ? theme.palette.warning.main
                    : "transparent",
              }}>
              <Typography
                variant="caption"
                sx={{
                  color,
                  display: "flex",
                  justifyContent: "cnter",
                  alignItems: "cnter",
                  flexDirection: "column",
                }}>
                {item.title}
                <Typography variant="caption" fontWeight={600} sx={{ color }}>
                  {item.value}
                </Typography>
              </Typography>
            </Box>
          );
        })}
      </Box>
      <Box
        sx={{
          display: "flex",
          flexDirection: "row",
          justifyContent: "space-around",
        }}>
        <Typography variant="h4" sx={{ color }}>
          23%
        </Typography>
        <Divider
          orientation="vertical"
          flexItem
          sx={{ backgroundColor: hexToRgba(color, 0.3) }}
        />
        <Typography
          variant="caption"
          sx={{
            color,
            display: "flex",
            justifyContent: "flex-start",
            alignItems: "flex-start",
            flexDirection: "column",
          }}>
          <Typography variant="caption" fontWeight={600} sx={{ color }}>
            -10%
          </Typography>
          Since last day
        </Typography>
      </Box>
    </Box>
  );
}
