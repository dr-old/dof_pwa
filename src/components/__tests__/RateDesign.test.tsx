// src/components/__tests__/RateDesign.test.tsx
import { render, screen, fireEvent } from "@testing-library/react";
import { ThemeProvider, createTheme } from "@mui/material/styles";
import RateDesign from "../RateDesign";
import "@testing-library/jest-dom";

// Create a custom theme to use in tests
const theme = createTheme({
  palette: {
    mode: "light",
    text: { primary: "#000" },
    background: { default: "#fff" },
    secondary: { main: "#ff5722" },
    warning: { main: "#ff9800" },
  },
});

const renderComponent = (props = {}) => {
  return render(
    <ThemeProvider theme={theme}>
      <RateDesign {...props} />
    </ThemeProvider>
  );
};

describe("RateDesign Component", () => {
  test("renders correctly", () => {
    const { asFragment } = renderComponent({
      title: "Test Title",
      backgroundColor: "#f0f0f0",
    });

    // Snapshot Test
    expect(asFragment()).toMatchSnapshot();

    // Check if title is rendered
    expect(screen.getByText("Test Title")).toBeInTheDocument();

    // Check if icon button is present
    expect(screen.getByRole("button")).toBeInTheDocument();

    // Check if the values are rendered
    expect(screen.getByText("Mon")).toBeInTheDocument();
    expect(screen.getByText("Tue")).toBeInTheDocument();
    expect(screen.getByText("Wed")).toBeInTheDocument();
  });

  test("handles click events", () => {
    renderComponent({ title: "Test Title", backgroundColor: "#f0f0f0" });

    // Click on the first item
    fireEvent.click(screen.getByText("Mon"));

    // Check if the border color changed (assuming border color changes on selection)
    const monBox = screen.getByText("Mon").closest("div");
    expect(monBox).toHaveStyle(`border-color: ${theme.palette.warning.main}`);
  });
});
