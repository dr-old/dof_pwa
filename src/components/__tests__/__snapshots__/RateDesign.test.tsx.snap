// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RateDesign Component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kl4l4i"
  >
    <div
      class="MuiBox-root css-1ialerq"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-12axgqo-MuiTypography-root"
      >
        Test Title
      </h6>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1azmo45-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall css-ptiqhd-MuiSvgIcon-root"
          data-testid="ArrowOutwardRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M6 7c0 .55.45 1 1 1h7.59l-8.88 8.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0L16 9.41V17c0 .55.45 1 1 1s1-.45 1-1V7c0-.55-.45-1-1-1H7c-.55 0-1 .45-1 1"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiBox-root css-1qt5u4s"
    >
      <div
        class="MuiBox-root css-gyvzpx"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-lniesh-MuiTypography-root"
        >
          Mon
          <span
            class="MuiTypography-root MuiTypography-caption css-1ihg57t-MuiTypography-root"
          >
            2
          </span>
        </span>
      </div>
      <div
        class="MuiBox-root css-gyvzpx"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-lniesh-MuiTypography-root"
        >
          Tue
          <span
            class="MuiTypography-root MuiTypography-caption css-1ihg57t-MuiTypography-root"
          >
            3
          </span>
        </span>
      </div>
      <div
        class="MuiBox-root css-gyvzpx"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-lniesh-MuiTypography-root"
        >
          Wed
          <span
            class="MuiTypography-root MuiTypography-caption css-1ihg57t-MuiTypography-root"
          >
            4
          </span>
        </span>
      </div>
    </div>
    <div
      class="MuiBox-root css-1vt08d7"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 css-1wq5wib-MuiTypography-root"
      >
        23%
      </h4>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-18t1r5k-MuiDivider-root"
      />
      <span
        class="MuiTypography-root MuiTypography-caption css-k7w3aq-MuiTypography-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-1ihg57t-MuiTypography-root"
        >
          -10%
        </span>
        Since last day
      </span>
    </div>
  </div>
</DocumentFragment>
`;
